### ISSUE TITLE
Client.get_collections_list throws type error

### ISSUE THREAD

**Issue by philvarner** [stac-utils/pystac-client#54]
Code:

```python
catalog: Client = Client.open('https://franklin.nasa-hsi.azavea.com/')
catalog.get_collections_list()
```

Error:

```
Traceback (most recent call last):
...
  File "/Users/philvarner/code/stac-api-validation-suite/stac_api_validation_suite/__init__.py", line 65, in validate_api
    for collection in catalog.get_collections_list():
  File "/Users/philvarner/.local/share/virtualenvs/stac-api-validation-suite-tzI1nfla/lib/python3.9/site-packages/pystac_client/client.py", line 160, in get_collections_list
    return self.get_child_links()
TypeError: get_child_links() missing 1 required positional argument: 'self'
```

Relevant code:

in client.py:
```
 @classmethod
    def get_collections_list(self):
        """Gets list of available collections from this Catalog. Alias for get_child_links since children
            of an API are always and only ever collections
        """
        return self.get_child_links()
```

calling this in pystac/catalog.py:

```
 def get_child_links(self):
        """Return all child links of this catalog.

        Return:
            List[Link]: List of links of this catalog with ``rel == 'child'``
        """
        return self.get_links('child')
```

**Comment by gadomski** [comment]
Duplicate of #44?

**Comment by philvarner** [comment]
yes, looks like it.

**Comment by matthewhanson** [comment]
Fixed in #45
